### PR TITLE
Fixed Issue 4346

### PR DESCRIPTION
**Pull request by paulkirow**
This is a fix for issue #4346 

The problem was that when handling parameters labelsize and labelcolor, _apply_params was calling setattr on _size and _color instead of _labelsize and _labelcolor.  This is all fine and dandy until something is used that calls _apply_params because _apply_params calls apply_tickdir which will change the padding.

When _size changes, its value gets added to the tick padding thus causing the issue seen in #4346.  This also explains why the issue only became apparent when setting major (or both) ticks, it is because minor ticks don't have _size added to their padding values.

### CODE REVIEW

**Comment by tacaswell** [comment]
Can you also add a test?  I would suggest a non-image test where you exercise this and assert that the tick size / color does not change due to it.  Ideally the tests should only use public methods.

Also :+1: on tracking this one down, this looks subtle as all get out.

**Comment by paulkirow** [comment]
I will work on this

**Comment by paulkirow** [comment]
Hopefully this is sufficient.

File: lib/matplotlib/tests/test_axes.py [hunk 1]
```diff
@@ -4335,6 +4335,17 @@ def test_pandas_indexing_hist():
...
     axes.hist(ser_2)
 
+@cleanup
+def test_axis_set_tick_params_labelsize_labelcolor():
+    # Tests fix for issue 4346
+    axis_1 = plt.subplot()
+    axis_1.yaxis.set_tick_params(labelsize=30, labelcolor='red', direction='out')
+
+    # Expected values after setting the ticks
+    assert axis_1.yaxis.majorTicks[0]._size == 4.0
```
Comment (tacaswell) [inline lib/matplotlib/tests/test_axes.py]: Is there any reliably way to test this using public API?

Comment (tacaswell) [inline lib/matplotlib/tests/test_axes.py]: The reason for asking that is that, long term, we reserve the right to change the private API with no warning or deprecation cycle.  Tests that only use public API give confidence that we have done that right but that is eroded if we have to touch the tests as part of the refactor as well.

Comment (paulkirow) [inline lib/matplotlib/tests/test_axes.py]: That makes sense, I didn't consider the public API.  I will look into this.

Comment (paulkirow) [inline lib/matplotlib/tests/test_axes.py]: I can't seem to find a way to get either of labelsize, labelcolor, size, or color out of the public API.  There is a [get_markersize](http://matplotlib.org/api/lines_api.html#matplotlib.lines.Line2D.get_markersize) in the docs that would return _size,  however I don't see it in the code. Should I implement this?

Comment (tacaswell) [inline lib/matplotlib/tests/test_axes.py]: If there is not a good public API, then this is fine.  Creating new API for the purpose of testing it is not a good idea!

**Comment by tacaswell** [comment]
backported to 1.5.x as b4c3091

**Comment by tacaswell** [comment]
Thanks!  And congratulations in what I think is your first mpl contribution :clap: 

In the future, it is best to make PRs off of a feature branch, not your master branch.

**Comment by paulkirow** [comment]
Will do, and thanks!

**Pull request merged**